### PR TITLE
APC code cleanup

### LINKED ISSUE #400: APC construction and deconstruction is broken
## Testmerges:
#366: 'Bug Fix Pack - 1-25-2019' by Surrealaser at commit 72c75fecb4 
#382: 'Admin Tweaks' by DominikPanic at commit bbd51aec01 
#397: 'Fixes simple mobs being catatonic and unkillable.' by Ghommie at commit d0145632c7 
#395: 'Quickfixes a few tarps problems.' by Ghommie at commit 201b32c879 
#394: 'Fixes larva being unable to go drone after queen's death.' by Ghommie at commit ecdd2ad78f 
#388: 'Lasguns can light cigarettes now' by Surrealaser at commit 9b62cf0407 
#385: 'Temporary Music PR (don't close me Qustinnus)' by DominikPanic at commit 24afbf25db 
#384: 'Adds a new anti-Brute medication called "Meralyne."' by Lainwright-Lazuli at commit 96e5936f21 
#368: 'Brig Overhaul' by Jimminian95 at commit f3c7b0498c 
#367: 'Re-ports Xeno Sprites + General Cleanup' by Alterist at commit e1e7920e10 
#320: 'Xeno Balance Tweak' by Surrealaser at commit b0585f87f1 

## Reproduction:

APC construction and deconstruction is bugged. You can build APC's until you need to apply cable coil, at which point no actions will proceed the construction. Using a screwdriver on the APC at this point leaves it in a bugged state where it appears to be functional with a cover, but provides no interface when clicked on.

APC deconstruction proceeds normally until you reach the stage after cutting the terminal. Using a crowbar to complete deconstruction gives an audio cue but nothing changes. Wirecutters can also be used to the same effect. The deconstruction can be backpedaled until the point where you would need to close the cover.

I don't build APC's often but the icon state of the APC when it gets placed onto the wall is a closed APC with a black screen. You can open the panel and it takes the normal exposed wiring state. Closing the panel gives it a red screen. 

Deleting the new bugged APC is odd. The deletion request seemingly takes a while to process. Deleting regular broken APC's processes instantly.

### PR DESCRIPTION
**Pull request by Menshin**
First pass of `power` code cleanup/update.
Lots of defines and code clarifications, readded a missing `New()` and a few missing cleanup (not removing from global list on `qdel()`, etc). I've kept the power logic mostly the same.

Get the `panel_open` var up to `obj/machinery` for future generalizations.

Fixes #400.

:cl: Menshin & spookydonut
fix: APC construction and deconstruction are working properly again
tweak: various APC code tweaking and updating 
/:cl:

### CODE REVIEW

**Comment by Rohesie** [comment]
Related: #413

**Comment by spookydonut** [comment]
See: https://github.com/Menshin/TerraGov-Marine-Corps/pull/1

File: code/modules/power/apc.dm [hunk 1]
```diff
@@ -133,38 +141,80 @@
...
+
+	if(mapload)
+		has_electronics = APC_ELECTRONICS_SECURED
+
+		//Is starting with a power cell installed, create it and set its charge level
+		if(cell_type)
+			cell = new cell_type(src)
+			cell.charge = start_charge * cell.maxcharge / 100.0 //Convert percentage to actual value
+
+		var/area/A = loc.loc
```
[the remark this answers is not in the document]
Comment (Menshin) [reply code/modules/power/apc.dm]: What the ... ?!? Probably reverted by accident. Fixing that in a moment.

**Comment by Rohesie** [comment]
Conflicts, please solve so we can merge this.

**Comment by Menshin** [comment]
Done.

**Pull request merged**